homeassistant/components/trello/config_flow.py

```diff
@@ -0,0 +1,220 @@
+        :return:
+        """
+        if user_input is None:
+            return await self._show_creds_form()
+
+        self.api_key = user_input[CONF_API_KEY]
+        self.api_token = user_input[CONF_API_TOKEN]
+        self.trello_adapter = _create_trello_adapter(
+            user_input[CONF_API_KEY], user_input[CONF_API_TOKEN]
+        )
+
+        try:
+            member = await self._get_current_member()
+        except Unauthorized as ex:
+            return await self._show_error_creds_form(ex)
+
+        self.user_id = member.id
+        self.user_email = member.email
+        self.ids_boards = await self._fetch_all_boards()
+
+        return await self._show_board_form()
+
+    async def async_step_finish(self, user_input: dict[str, Any]) -> FlowResult:
+        """Select desired boards to have card counts of per list.
+
+        :param user_input: User's selected boards
+        """
+        boards: dict[str, dict] = await self._get_boards_lists(
+            user_input[USER_INPUT_BOARD_IDS]
+        )
+
+        await self.async_set_unique_id(self.user_id)
+        self._abort_if_unique_id_configured()
+
+        config_data: dict[str, str] = self._get_config_data()
+        config_options = {CONF_OPTIONS_BOARDS: boards}
+
+        return self.async_create_entry(
+            title=self.user_email, data=config_data, options=config_options
+        )
+
+    def _get_config_data(self) -> dict[str, str]:
+        return {
+            CONF_API_KEY: self.api_key,
+            CONF_API_TOKEN: self.api_token,
+            CONF_USER_ID: self.user_id,
+            CONF_USER_EMAIL: self.user_email,
+        }
+
+    async def _show_creds_form(self) -> FlowResult:
+        return self.async_show_form(
+            step_id="user", data_schema=CREDS_FORM_SCHEMA, last_step=False
+        )
+
+    async def _show_board_form(self) -> FlowResult:
+        return self.async_show_form(
+            step_id="finish",
+            data_schema=_get_board_select_schema(self.ids_boards),
+            last_step=True,
+        )
+
+    async def _show_error_creds_form(self, ex: Unauthorized) -> FlowResult:
+        LOGGER.error("Unauthorized: %s)", ex)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=CREDS_FORM_SCHEMA,
+            errors={"base": "invalid_auth"},
+            last_step=False,
+        )
+
+    async def _fetch_all_boards(self) -> dict[str, dict[str, str]]:
+        return await self.hass.async_add_executor_job(self.trello_adapter.get_boards)
+
+    async def _get_boards_lists(self, board_ids: list[str]) -> dict[str, dict]:
+        return await self.hass.async_add_executor_job(
+            self.trello_adapter.get_board_lists, self.ids_boards, board_ids
+        )
+
+    async def _get_current_member(self) -> Member:
+        return await self.hass.async_add_executor_job(self.trello_adapter.get_member)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
I'd suggest moving this to a separate PR. Also, if you're continuing to work on it in the meantime, checkout OptionsFlowWithConfigEntry (check the lastfm one how it is used)